### ISSUE TITLE
Merge early work on "Recommended compiler option flags for C/C++ programs" into latest draft

### ISSUE THREAD

**Issue by david-a-wheeler** [ossf/wg-best-practices-os-developers#97]
There are many programs written in C or C++. Compiler option flags can help reduce some of their risks, but to my knowledge there's no common place for that.

We had started creating one here:

"Recommended compiler option flags for C/C++ programs"
https://docs.google.com/document/d/1SslnJuqbFUyTFnhzkhC_Q3PPGZ1zrG89COrS6LV6pz4/edit

It's still in early draft form. Should we work on this further?

**Comment by mrybczyn** [comment]
Interested in this one

**Comment by gkunz** [comment]
Hi @SecurityCRob,

what do you think about putting this topic on the agenda for the next meeting? I'd like to bring a colleague who is interested in contributing content to the guide. He could walk us through some content we already have.

**Comment by SecurityCRob** [comment]
I LOVE the idea!  I’ll get the next agenda setup tomorrow, feel free to add his discussion once it is ready!

Cheers,

CRob
Director of Security Communications
Intel Product Assurance and Security

**Comment by thomasnyman** [comment]
I opened a draft PR ( https://github.com/ossf/wg-best-practices-os-developers/pull/113 ) with the initial Markdown draft of the "Toolchain hardening for C and C++" guide discussed during the C/C++ Compiler Options BP Guide call on 2023-02-15 in order to make it available for review and feedback.

**Comment by david-a-wheeler** [comment]
All: The work contributed by Ericsson is far more complete, and thus is a better starting point. So I've changed the issue title.

This issue is now focused on taking our earlier work titled
["Compiler Options Hardening Guide for C and C++"](https://docs.google.com/document/d/1SslnJuqbFUyTFnhzkhC_Q3PPGZ1zrG89COrS6LV6pz4/edit#) and merge information that only it has into our current draft at <https://github.com/ossf/wg-best-practices-os-developers/blob/main/docs/Compiler_Hardening_Guides/Compiler-Options-Hardening-Guide-for-C-and-C%2B%2B.md>

**Comment by SecurityCRob** [comment]
Has this been superseded by the C/C++ Compiler Hardening options guide?

**Comment by SecurityCRob** [comment]
@gkunz @thomasnyman @david-a-wheeler

**Comment by david-a-wheeler** [comment]
We're done, thank you everyone!
